### PR TITLE
MAGN-7724 Seperate out Revit dependancies from current Dynamo implementation of Solar Radiation

### PR DESCRIPTION
**Pull request by ikeough**
### Purpose

This PR addresses [MAGN-7724](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-7724). 

Most of the work for MAGN-7724 is done in the Dynamo-Solar-Analysis repository. The reviewer should review [this PR](https://git.autodesk.com/keoughi/Dynamo-Solar-Analysis/pull/2) as part of the current PR.

In the Dynamo repository, we've added `Date Time` node, which takes a formatted date and time as input and creates a `DateTime` object.  

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] The level of testing this PR includes is appropriate
  - New tests were added against the serialization/deserialization of the Date Time input node.
- [x] User facing strings, if any, are extracted into `*.resx` files

### Reviewers

@pboyer

### CODE REVIEW

File: src/DynamoSandbox/app.config [hunk 1]
```diff
@@ -8,8 +8,32 @@
       <probing privatePath="dll"/>
     </assemblyBinding>
   </runtime>
+  <system.serviceModel>
```
Comment (ikeough) [inline src/DynamoSandbox/app.config]: Config data for the Thor climate server is added here. This is the same as is found in the Revit app config.

File: src/Libraries/CoreNodeModelsWpf/Converters/StringToDateTimeOffsetConverter.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Globalization;
+using System.Windows.Data;
+
+namespace Dynamo.Wpf.Converters
+{
+    public class StringToDateTimeOffsetConverter : IValueConverter
```
Comment (ikeough) [inline src/Libraries/CoreNodeModelsWpf/Converters/StringToDateTimeOffsetConverter.cs]: This converter converts a date/time string into a `DateTime` object. If conversion fails, it returns a new `DateTime` object.

File: src/Libraries/CoreNodeModels/Input/DateTime.cs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+namespace DSCoreNodesUI
+{
+    [NodeName("Date Time")]
+    [NodeDescription("Create a DateTime object by specifying a date and a time.")]
+    [NodeCategory(BuiltinNodeCategories.CORE_INPUT)]
+    [IsDesignScriptCompatible]
+    public class DateTime : BasicInteractive<System.DateTime>
+    {
+        public DateTime()
```
Comment (ikeough) [inline src/Libraries/CoreNodeModels/Input/DateTime.cs]: The `DateTime` class wraps the creation of a `DateTime` object.

File: src/DynamoSandbox/app.config (hunk 1 above)
Comment (pboyer) [inline src/DynamoSandbox/app.config]: LOL.  Thor!

File: src/Libraries/CoreNodeModels/Input/DateTime.cs (hunk 3 above)
Comment (pboyer) [inline src/Libraries/CoreNodeModels/Input/DateTime.cs]: Wait a minute, don't we have this already?  

https://github.com/DynamoDS/Dynamo/blob/6feb670f9bc64c9d6048ac297d0ce30eab1ab01e/src/Libraries/CoreNodes/DateTime.cs

File: src/Libraries/CoreNodeModelsWpf/Controls/DateTimeInputControl.xaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+             xmlns:d="http://schemas.microsoft.com/expression/blend/2008" 
+             xmlns:converters="clr-namespace:Dynamo.Wpf.Converters"
+             mc:Ignorable="d" 
+             Width="200">
+    <UserControl.Resources>
+        <converters:StringToDateTimeOffsetConverter x:Key="StringToDateTimeOffsetConverter"/>
+    </UserControl.Resources>
+    <Grid>
+        <TextBox Text="{Binding Value, Mode=TwoWay, Converter={StaticResource StringToDateTimeOffsetConverter}}"
+                     ToolTip="Ented time and date in the format 'dd MMMM yyyy h:mm tt'"
```
Comment (pboyer) [inline src/Libraries/CoreNodeModelsWpf/Controls/DateTimeInputControl.xaml]: Shouldn't this be in a resource file?  Also there's a typo.

File: src/Libraries/CoreNodeModels/Input/DateTime.cs (hunk 3 above)
Comment (pboyer) [inline src/Libraries/CoreNodeModels/Input/DateTime.cs]: I guess you're trying to make some sort of custom UI around it.  Forget it...

File: src/AssemblySharedInfoGenerator/AssemblyInfoGenerator.csproj [hunk 5]
```diff
@@ -36,13 +36,17 @@
...
+    <Reference Include="PresentationFramework" />
     <Reference Include="System" />
     <Reference Include="System.Core" />
+    <Reference Include="System.Xaml" />
     <Reference Include="System.Xml.Linq" />
     <Reference Include="System.Data.DataSetExtensions" />
     <Reference Include="Microsoft.CSharp" />
     <Reference Include="System.Data" />
     <Reference Include="System.Xml" />
+    <Reference Include="WindowsBase" />
```
Comment (pboyer) [inline src/AssemblySharedInfoGenerator/AssemblyInfoGenerator.csproj]: Alarms are going off in my head. Why are we making this change?

File: src/Libraries/CoreNodeModels/Input/DateTime.cs [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        protected override System.DateTime DeserializeValue(string val)
+        {
+            System.DateTime result;
+            return System.DateTime.TryParseExact(val, PreferenceSettings.DefaultDateFormat, CultureInfo.InvariantCulture, DateTimeStyles.None, out result) ? result : new System.DateTime();
+        }
+
+        protected override string SerializeValue()
+        {
+            return Value.ToString(PreferenceSettings.DefaultDateFormat, CultureInfo.InvariantCulture);
+        }
```
Comment (pboyer) [inline src/Libraries/CoreNodeModels/Input/DateTime.cs]: We need some tests to confirm users can save and reopen workspaces with this node.

File: src/Libraries/CoreNodeModels/Input/DateTime.cs (hunk 3 above)
Comment (ikeough) [inline src/Libraries/CoreNodeModels/Input/DateTime.cs]: @pboyer You are correct. This bit of UI is out of scope. @lillismith and I have discussed this. This UI is a first pass implementation until we implement a better date picker control. We needed to have SOMETHING, that wasn't the user constructing a date time object with a minimum of 5 nodes.

File: src/AssemblySharedInfoGenerator/AssemblyInfoGenerator.csproj (hunk 5 above)
Comment (ikeough) [inline src/AssemblySharedInfoGenerator/AssemblyInfoGenerator.csproj]: Not sure why this got added. I've removed it, and all the other unused references in that project.

**Comment by ikeough** [comment]
@pboyer I might need your help with AppVeyor here. The resource file is available. Everything builds locally. Are there special settings to adjust for AppVeyor?

**Comment by ikeough** [comment]
@pboyer I've attached links in the PR description for the internal repo version of the work. Please let me know if you are unable to review.

**Comment by pboyer** [comment]
LGTM, but looks like we have some conflicts.

**Pull request merged**